### PR TITLE
[IMP] spreadsheet_edition: allow insertion of list selection

### PR DESCRIPTION
**Pull request by khpa-odoo**
This PR introduces the functionality to allows users to insert selective records from the list view.

The functionality is achieved by adding a new domain alongwith the
existing 'search model domain', via `getListForSpreadsheet` method.

This new domain filters the selected records using 'resIds' and thus
enables the user to insert only the selected records from current search.

Task ID : 3487810

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/tests/mobile/views/list_view_tests.js [hunk 1]
```diff
@@ -85,9 +85,9 @@ QUnit.module("Mobile Views", ({ beforeEach }) => {
         assert.containsOnce(fixture, "div.o_control_panel .o_cp_action_menus");
 
         await toggleActionMenu(fixture);
-        assert.deepEqual(
-            getMenuItemTexts(fixture.querySelector(".o_cp_action_menus")),
-            ["Duplicate", "Delete"],
+        const menuItemTexts = getMenuItemTexts(fixture.querySelector(".o_cp_action_menus"));
+        assert.ok(
+            ["Duplicate", "Delete"].every((menuitem) => menuItemTexts.includes(menuitem)),
```
Comment (FrancoisGe) [inline addons/web/static/tests/mobile/views/list_view_tests.js:90]: Why do you need to edit tests, without any change in web/ ??
If you do a patch in enterprise, you should apply it only in your test suite.

Comment (rrahir) [reply addons/web/static/tests/mobile/views/list_view_tests.js:90]: following her patch ,the number of default 'action items' will change based on if the module is installed or not. We patch a very generic menu in Enterprise, the strict equality of the test does not account for such a change. Do you have a suggestion?

Comment (FrancoisGe) [reply addons/web/static/tests/mobile/views/list_view_tests.js:90]: It s only a js patch ? (not in xml ?) If it s just JS, you can apply the patch only where you need it

Comment (FrancoisGe) [reply addons/web/static/tests/mobile/views/list_view_tests.js:90]: But not in web/tests/, you want to do something like in /community/addons/hr/static/tests/disable_patch.js. To unpatch it in the test and apply it only in your test suite :)

**Pull request closed without merging**